src/components/manage/Blocks/Title/View.jsx

```diff
@@ -3,19 +3,29 @@
  * @module volto-slate/blocks/Title/TitleBlockView
  */
 
-import React from 'react';
+import React, { useMemo } from 'react';
 import PropTypes from 'prop-types';
+import Slugger from 'github-slugger';
+import { renderLinkElement } from '@plone/volto-slate/editor/render';
 
 /**
  * View title block component.
  * @class View
  * @extends Component
  */
-const TitleBlockView = ({ properties, metadata }) => {
+const TitleBlockView = ({ properties, metadata, id, children }) => {
+  let attr = { id };
+  const title = (properties || metadata)['title'];
+  const slug = Slugger.slug(title);
+  attr.id = slug || id;
+  const LinkedTitle = useMemo(() => renderLinkElement('h1'), []);
```

Comment:
Just for my understanding: What's the effect of useMemo here?

Reply:
@ksuess 

Let's say the code is like this:

```
const LinkedTitle = renderLinkElement('h1');
```

The renderLinkElements, whenever it is called, builds a new component. If for some reason the Title > View component is refreshed and runs again, the LinkedTitle would become a new component and react would unmount the old component `<LinkedTitle>` and create a new one instead. The `useMemo` is in fact a memoization, so that whenever the `View` component runs, it gets the same `LinkedTitle` component and so, when it's rendered, doesn't cause unnecessary unmount. In this case React would reuse the already mounted `<LinkedTitle>` and, if some prop changed (for example, the title was updated from the content), the component would receive the new prop (and, in class-based component terminology, would do a `componentDidUpdate`, while in function component, it would trigger any of its `React.useEffect()`.

Reply:
Got it. Thank you @tiberiuichim !